### PR TITLE
add environment tests

### PR DESCRIPTION
**Pull request by petrkalos**
### Feature or Bugfix
Feature

### Detail
* add list_environment tests
* add test for updating an environment (via update_stack)
* generalise the polling functions for stacks

### Relates
#1220 

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests_new/integration_tests/core/environment/global_conftest.py [hunk 1]
```diff
@@ -87,5 +86,21 @@ def temp_env1(client1, group1, org1, testdata):
...
-def persistent_env1(client1, group1, org1, testdata): ...  # TODO
+def get_or_create_persistent_env(env_name, client, group, testdata):
+    envs = list_environments(client, term=env_name).nodes
+    if envs:
+        return envs[0]
+    else:
+        envdata = testdata.envs[env_name]
+        org = create_organization(client, f'org_{env_name}', group)
+        env = create_env(client, group, org['organizationUri'], envdata.accountId, envdata.region, tags=[env_name])
+        if env.stack.status == 'CREATE_COMPLETE':
```
Comment (dlpzx) [inline tests_new/integration_tests/core/environment/global_conftest.py:97]: What about UPDATE_COMPLETE?

Comment (dlpzx) [reply tests_new/integration_tests/core/environment/global_conftest.py:97]: Every day there is an scheduled task that updates environment and dataset stacks. It is an edge case that exactly at the same time we run our tests, but it could happen. Should we consider it in the design?

**Review by dlpzx — COMMENTED** [review]
Left a small comment, but it looks great

File: tests_new/integration_tests/core/environment/global_conftest.py (hunk 1 above)
Comment (petrkalos) [reply tests_new/integration_tests/core/environment/global_conftest.py:97]: Although this will be pretty unlikely, polling is 10seconds, but it won't harm adding it. Will also have to add it to the assertions.

**Review by dlpzx — APPROVED** [review]
